app/src/main/java/com/keylesspalace/tusky/components/viewthread/ViewThreadViewModel.kt

```diff
@@ -277,27 +276,14 @@ class ViewThreadViewModel @Inject constructor(
         }
     }
 
-    private fun handleFavEvent(event: FavoriteEvent) {
-        updateStatus(event.statusId) { status ->
-            status.copy(favourited = event.favourite)
-        }
-    }
-
-    private fun handleReblogEvent(event: ReblogEvent) {
-        updateStatus(event.statusId) { status ->
-            status.copy(reblogged = event.reblog)
-        }
-    }
-
-    private fun handleBookmarkEvent(event: BookmarkEvent) {
-        updateStatus(event.statusId) { status ->
-            status.copy(bookmarked = event.bookmark)
-        }
-    }
-
-    private fun handlePinEvent(event: PinEvent) {
-        updateStatus(event.statusId) { status ->
-            status.copy(pinned = event.pinned)
+    private fun handleStatusChangedEvent(status: Status) {
+        updateStatusViewData(status.id) { viewData ->
+            status.toViewData(
+                isShowingContent = viewData.isShowingContent,
+                isExpanded = viewData.isExpanded,
+                isCollapsed = viewData.isCollapsed,
+                isDetailed = viewData.isDetailed
+            )
         }
     }
 
```

Comment:
Do you need to update `handleStatusComposedEvent()` as well?

Specifically, if the status that was composed is a reply, the status it was replying to needs to be updated so that the reply count is correct.

Reply:
yes that should probably be done, but I don't see it in the scope of this PR